src/registry/customerProfile/validationSchema.ts

```diff
@@ -0,0 +1,172 @@
+import type {
+  CustomerProfileComponentSchema,
+  DigitalAddress,
+  DigitalAddressType,
+} from '@open-formulieren/types';
+import type {IntlShape} from 'react-intl';
+import {defineMessage} from 'react-intl';
+import {z} from 'zod';
+
+import type {GetValidationSchema} from '@/registry/types';
+
+const USE_ONLY_ONCE_AND_AS_PREFERRED_CONSTRAINT_MESSAGE = defineMessage({
+  description:
+    'Validation error for customerProfile digital address useOnlyOnce and isNewPreferred that are both true',
+  defaultMessage: "The digital address cannot be marked as both 'one-time use' and 'preferred'.",
+});
+
+const DUPLICATE_DIGITAL_ADDRESS_TYPES_MESSAGE = defineMessage({
+  description:
+    'Validation error for customerProfile with multiple digital addresses for the same type',
+  defaultMessage: 'You cannot submit multiple digital addresses for the type {digitalAddressType}.',
+});
+
+const EMAIL_INVALID_MESSAGE = defineMessage({
+  description:
+    'Validation error for customerProfile digital address that does not match the email format',
+  defaultMessage: 'Invalid email address.',
+});
+
+const PHONE_NUMBER_INVALID_MESSAGE = defineMessage({
+  description:
+    'Validation error for customerProfile digital address that does not match the phone number format',
+  defaultMessage: 'Invalid phone number.',
+});
+
+/**
+ * Builds a digital address validation schema for a specific digital address `type` and
+ * `address` schema.
+ *
+ * This is a helper function used by the digital address type-specific builder functions.
+ * Using the provided `basicAddressSchema`, this function builds a validation schema for
+ * the specified `type`. This is not intended to be used directly by the
+ * getValidationSchema or customerProfile component registry.
+ *
+ * The generic rules for digital addresses are defined in this builder. These include:
+ * - The `type` field must be one of the allowed types for the digital address.
+ * - The `address` is optional or required based on the component `required` parameter.
+ *
+ * @param required - `required` parameter from the parent component.
+ * @param type - For which type of digital address the schema is meant.
+ * @param addressSchema - The validation schema for the address field.
+ */
+const buildDigitalAddressSchema = (
+  required: boolean,
```

Comment:
we tend to prefix booleans with `is` or `has`


```suggestion
  isRequired: boolean,
```

it reads very natural in JSX when used as bare prop: `<Component isRequired />`, but in general it's a good naming pattern.